borg/archiver.py

```diff
@@ -177,6 +178,7 @@ def do_check(self, args, repository):
     @with_repository()
     def do_change_passphrase(self, args, repository, manifest, key):
         """Change repository key file passphrase"""
+        check_repository_compatibility(manifest, REPO_OPERATION_CHECK)
```

Comment:
having to add that everywhere to avoid damage seems error prone, it could be forgotten.
so, add it to the decorator maybe?